### PR TITLE
Update add_overlay to make use of the full pycoast capabilities

### PR DESCRIPTION
**Pull request by mraspaud**
This requires https://github.com/pytroll/pycoast/pull/32 and https://github.com/pytroll/trollimage/pull/57 to be merged and released

 - [x] Tests added and test suite added to parent suite 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by mraspaud** [comment]
After thinking more about this, I think `trollimage.XRImage` should expose a method called something like `pil_apply` that allows the user to apply some pillow operations on the image. This way, satpy wouldn't need to make any assumption on how an XRImage works, so we could call for example
```python
new_img = orig_img.pil_apply(_burn_overlays)(*args, **kwargs)
```
(`_burn_overlays` would have to be modify of course)
The delaying of the pil operations would then be done in trollimage.

File: satpy/writers/__init__.py [hunk 1]
```diff
@@ -232,56 +236,46 @@ def add_overlay(orig, area, coast_dir, color=(0, 0, 0), width=0.5, resolution=No
...
-    # Only add grid if major_lonlat is given.
-    if grid and 'major_lonlat' in grid and grid['major_lonlat']:
-        major_lonlat = grid.pop('major_lonlat')
-        minor_lonlat = grid.pop('minor_lonlat', major_lonlat)
 
-        cw_.add_grid(img, area, major_lonlat, minor_lonlat, **grid)
+    if overlays is None:
+        # fill with sensible defaults
+        general_params = {'outline': color or (0, 0, 0),
+                          'width': width or 0.5}
```
Comment (djhoese) [inline satpy/writers/__init__.py]: Why are these sensible defaults not the defaults in pycoast?

File: satpy/writers/__init__.py [hunk 2]
```diff
@@ -232,56 +236,46 @@ def add_overlay(orig, area, coast_dir, color=(0, 0, 0), width=0.5, resolution=No
...
-    new_data = xr.DataArray(arr, dims=['y', 'x', 'bands'],
-                            coords={'y': orig.data.coords['y'],
-                                    'x': orig.data.coords['x'],
-                                    'bands': list(img.mode)},
-                            attrs=orig.data.attrs)
-    return XRImage(new_data)
+    cw_ = ContourWriterAGG(coast_dir)
+    new_image = orig_img.apply_pil(_burn_overlay, res_mode,
+                                   None, {'fill_value': fill_value},
+                                   (area, cw_, overlays), None)
```
Comment (djhoese) [inline satpy/writers/__init__.py]: Is it possible to provide backwards compatibility for a while or is it too much work?

File: satpy/writers/__init__.py (hunk 1 above)
Comment (mraspaud) [reply satpy/writers/__init__.py]: Haha good question :) I just took what we had before in this method

File: satpy/writers/__init__.py (hunk 2 above)
Comment (mraspaud) [reply satpy/writers/__init__.py]: Should be doable, let's see

Comment (mraspaud) [reply satpy/writers/__init__.py]: ok, I can do this (copy apply_pil in satpy), but it's ugly and doesn't feel good as it's just a temporary patch.
Moreover, we also need a new version of pycoast for this to work.
So if you really want to keep backwards compatibility, I could just keep the old `add_overlay` and add a call to the new one inside in instead. But do we really need this ?

Comment (djhoese) [reply satpy/writers/__init__.py]: > But do we really need this?

"Probably not" -Famous last words

Users are going to complain, but it seems like it is much easier on us to just say "update pycoast".

File: .travis.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ env:
         - PYTHON_VERSION=$TRAVIS_PYTHON_VERSION
         - NUMPY_VERSION=stable
         - MAIN_CMD='python setup.py'
-        - CONDA_DEPENDENCIES='xarray!=0.13.0 dask distributed toolz Cython sphinx cartopy pillow matplotlib scipy pyyaml pyproj pyresample coveralls coverage codecov behave netcdf4 h5py h5netcdf gdal rasterio imageio pyhdf mock libtiff pycoast pydecorate geoviews zarr six'
+        - CONDA_DEPENDENCIES='xarray!=0.13.0 dask distributed toolz Cython sphinx cartopy pillow matplotlib scipy pyyaml pyproj pyresample coveralls coverage codecov behave netcdf4 h5py h5netcdf gdal rasterio imageio pyhdf mock libtiff geoviews zarr six'
```
Comment (djhoese) [inline .travis.yml]: Are these being implicitly installed with the test commands? Shouldn't these be added to the PIP_DEPENDENCIES?

Comment (mraspaud) [reply .travis.yml]: No, they are mocked now, so they aren't needed anymore fro the tests

Comment (djhoese) [reply .travis.yml]: Oh interesting. Do we want them to be mocked?

**Comment by mraspaud** [comment]
I'll make an issue, thanks for reviewing so fast !

File: .travis.yml (hunk 3 above)
Comment (mraspaud) [reply .travis.yml]: I think for unit tests they should. But I also think we should have system tests (running every night) what would use actual pycoast and pydecorate to make sure compatibility isn't broken.
IMHO this has the following advantages: 
- we can continue having passing tests (and thus continue developping) even when an external lib changes its API
- unit tests install and run faster
- unit tests don't need to know the internals of the external lib, so it keeps the testing code cleaner
BUT as said above, we then need to run system tests to make sure the whole system still works.

**Pull request merged**